#!/usr/bin/env python
PACKAGE = "youbot_kinematics_rviz"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

#       Name    Type    Level 	Description           Default    Min   Max
gen.add("arm1", double_t, 0, "An double degree parameter", 0, -169, 169)
gen.add("arm2", double_t, 0, "An double degree parameter", 0, -65, 90)
gen.add("arm3", double_t, 0, "An double degree parameter", 0, -151, 146)
gen.add("arm4", double_t, 0, "An double degree parameter", 0, -102.5, 102.5)
gen.add("arm5", double_t, 0, "An double degree parameter", 0, -165, 165)

gen.add("gripper_l_param", double_t, 0, "An double parameter unit meter", 0.008, 0, 0.0115)
gen.add("gripper_r_param", double_t, 0, "An double parameter unit meter", 0.008, 0, 0.0115)


exit(gen.generate(PACKAGE, "youbot_kinematics_rviz", "arm_joint_state"))
